commands/scale/cluster/command.go

```diff
@@ -54,296 +52,255 @@ Examples:
-			DesiredScalingMaxChanged: &desiredScalingMaxChanged,
-			DesiredScalingMin:        &desiredScalingMin,
-			DesiredScalingMinChanged: &desiredScalingMinChanged,
-		}
-
-		scaling, err = defaulting.NewScaling(c)
-		if err != nil {
-			return Arguments{}, microerror.Mask(err)
-		}
+	if args.Workers > 0 {
+		args.WorkersMin = int64(args.Workers)
+		args.WorkersMax = int64(args.Workers)
 	}
 
-	req := request.Request{}
-
-	req.Cluster.Scaling = scaling.Default(ctx, req.Cluster.Scaling)
+	return args, nil
+}
 
-	scaleArgs.workersMax = req.Cluster.Scaling.Max
-	scaleArgs.workersMin = req.Cluster.Scaling.Min
+func verifyPreconditions(args Arguments) error {
+	if args.AuthToken == "" && args.UserProvidedToken == "" {
+		return microerror.Mask(errors.NotLoggedInError)
+	}
+	if args.ClusterID == "" {
+		return microerror.Mask(errors.ClusterIDMissingError)
+	}
+	if args.WorkersSet && (args.WorkersMinSet || args.WorkersMaxSet) {
+		return microerror.Mask(errors.ConflictingWorkerFlagsUsedError)
+	}
+	if args.WorkersMax > 0 && args.WorkersMax < int64(limits.MinimumNumWorkers) {
+		return microerror.Mask(errors.CannotScaleBelowMinimumWorkersError)
+	}
+	if args.WorkersMin > 0 && args.WorkersMin < int64(limits.MinimumNumWorkers) {
+		return microerror.Mask(errors.NotEnoughWorkerNodesError)
+	}
+	if args.Workers != 0 && args.Workers < limits.MinimumNumWorkers {
+		return microerror.Mask(errors.NotEnoughWorkerNodesError)
+	}
+	if !args.WorkersSet && !args.WorkersMinSet && !args.WorkersMaxSet {
+		return microerror.Maskf(errors.RequiredFlagMissingError, "--%s or --%s/--%s", cmdWorkersNumName, cmdWorkersMinName, cmdWorkersMaxName)
+	}
 
-	return scaleArgs, nil
+	return nil
 }
 
-// getClusterStatus returns the status for one cluster.
-func getClusterStatus(clusterID, activityName string) (*client.ClusterStatus, error) {
-	clientWrapper, err := client.NewWithConfig(flags.APIEndpoint, flags.Token)
+func printValidation(cmd *cobra.Command, positionalArgs []string) {
+	args, err := collectArguments(cmd, positionalArgs)
 	if err != nil {
-		return nil, microerror.Mask(err)
+		fmt.Println(color.RedString(err.Error()))
+		os.Exit(1)
 	}
 
-	auxParams := clientWrapper.DefaultAuxiliaryParams()
-	auxParams.ActivityName = activityName
-
-	// Make sure we have provider info in the current endpoint
-	if config.Config.Provider == "" {
-		if flags.Verbose {
-			fmt.Println(color.WhiteString("Fetching provider information"))
-		}
+	err = verifyPreconditions(args)
 
-		info, err := clientWrapper.GetInfo(auxParams)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-		config.Config.SetProvider(info.Payload.General.Provider)
+	if err == nil {
+		return
 	}
 
-	// perform API call
-	if flags.Verbose {
-		fmt.Println(color.WhiteString("Fetching current cluster size"))
-	}
-	status, err := clientWrapper.GetClusterStatus(clusterID, auxParams)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
+	errors.HandleCommonErrors(err)
 
-	return status, nil
-}
+	headline := ""
```

Comment:
```suggestion
	var headline string
```
Extremely minor thing, but something that I noticed thinking. Given our usual practice to have blocks like following:
```go
var foo TypeBar
{
    f, err := findOutFoo()
    if err != nil {
        return microerror.Mask(err)
    }
    foo = f
}
```

It's somehow easier to anticipate following logic when the variable is just declared instead of introduced and explicitly initialized when in here the actually value assignation is done in following block. No strong opinions here but something that I found myself thinking about.